modules/index.js

```diff
@@ -1296,7 +1296,7 @@ export function isEqual(a, b) {
 // An "empty" object has no enumerable own-properties.
 export function isEmpty(obj) {
   if (obj == null) return true;
-  if (isArrayLike(obj) && (isArray(obj) || isString(obj) || isArguments(obj))) return obj.length === 0;
+  if (isArray(obj) || isString(obj) || isArguments(obj)) return obj.length === 0;
```

Comment:
This wasn’t a tautology, it was an optimization! We only wanted to proceed to the three `toString()` equality checks on the right if we were sure that we already had a numeric length property on the left.

That said, it may only have been an optimization for ancient versions of IE. You’d have to measure to determine if it matters at all anymore.

If you want to leave the optimization in, perhaps add a comment, eg: `// Only proceed to check types if a numeric .length property is present.`

Reply:
Ah, good thing I asked for a review. So it's still a tautology but *also* an optimization. ;-)

I would do some profiling, but I can't test in IE11 or Edge. So I think I'll probably just leave it in and go for the comment option.

Reply:
Testing anyway.

Reply:
Alright. I'm quite sure I totally overdid this, but since I took the effort, here are the conclusions of my testing.

On Node 10, `isArrayLike` and `isArray` are about equally cheap operations.

On Firefox 75, `isArray` takes nearly three times as much time as `isArrayLike`.

On Safari 13, `isArray` takes about 18 times as much time as `isArrayLike`.

Overall, it definitely seems better to keep the optimization even if you don't take old IE into account.

I ensured maximum optimization by running the hot loop millions of times on each trial and used Latin squares and systems of linear equations in order to obtain accurate numbers that could be compared. Impressively, Safari needs only about 230 CPU cycles to make a single iteration in its best scenario (plain object with optimization). Firefox can do the array without optimization scenario in about 245 CPU cycles. Taking the `length` of a string is much more expensive than taking the `length` of an array or `arguments`, perhaps because a string has to be object-wrapped first.

I don't really expect anyone to care, but script and data [here](https://gist.github.com/jgonggrijp/06644e2667466beaa32324b7b066e6a2).

Tomorrow, I'll put back the optimization and add a comment as you suggested.

Reply:
Very cool! Thanks for the benchmarks!